lib/puppet/util/windows/adsi.rb

```diff
@@ -463,13 +465,11 @@ def remove_member_sids(*sids)
       end
     end
 
+    # this now returns SID objects instead of strings
     def members
-      member_sids.map(&:domain_account)
-    end
-
-    def member_sids
       self.class.get_sids(native_group.Members)
     end
+    alias member_sids members
```

Comment:
This is the part I'm most concerned about changing. `members` used to be a list of strings like `['NT AUTHORITY\SYSTEM', 'Administrators']`, etc... but now it's an array of `Principal` objects.  Our hands were a bit tied here as Puppet internals will call `property.retrieve` against `members`, which is what sets the `current` value... and we wanted to preserve information that was already in hand.

Another option would have been to return things as they were previously returned - using SID strings like `S-1-5-XXX` when they couldn't be resolved to a name. Then the provider would have to understand that it must ignore them when they fail to resolve back to SIDs. That would end up being a bit silly since we already know at this point whether the SID is resolvable or not.  In other words we can avoid the bytes from ADSI -> lookup SID bytes to synthesize domain prefixed account -> lookup name to get SID, which introduces extraneous work.

Because `members` is now a list of `Principal` objects, this impacts `puppet resource`, which typically calls `to_s` when rendering values. To make sure generated manifests use friendly names instead of SIDs, the `to_s` in Principal has to be changed accordingly.